jlm/llvm/ir/operators/lambda.hpp

```diff
@@ -663,7 +669,197 @@ class node::fctresconstiterator final : public jlm::rvsdg::input::constiterator<
   }
 };
 
-}
+/**
+ * The CallSummary of a lambda summarizes all call usages of the lambda. It distinguishes between three call usages:
+ *
+ * 1. The export of the lambda, which is null if the lambda is not exported.
+ * 2. All direct calls of the lambda.
+ * 3. All other usages, e.g., indirect calls.
+ */
+class node::CallSummary final
+{
+  using DirectCallsConstRange = util::iterator_range<std::vector<CallNode*>::const_iterator>;
+  using OtherUsersConstRange = util::iterator_range<std::vector<rvsdg::simple_input*>::const_iterator>;
+
+public:
+  CallSummary(
+    rvsdg::result * rvsdgExport,
+    std::vector<CallNode*> directCalls,
+    std::vector<rvsdg::simple_input*> otherUsers)
+    : RvsdgExport_(rvsdgExport)
+    , DirectCalls_(std::move(directCalls))
+    , OtherUsers_(std::move(otherUsers))
+  {}
+
+  CallSummary(const CallSummary&) = delete;
```

Comment:
copy & move of the class would be "safe", so it would not be necessary to be so strict here

Reply:
Removed explicit deletion.